File: src/main/java/itmo/is/project/repository/user/UserRepository.java
package itmo.is.project.repository.user;

import itmo.is.project.model.user.Role;
import itmo.is.project.model.user.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Integer> {
    boolean existsByUsername(String username);

    boolean existsByRole(Role role);

    Optional<User> findByUsername(String username);

    Page<User> findAllByEnabledFalse(Pageable pageable);
}

File: src/main/kotlin/itmo/isproject/repository/user/UserRepository.kt
package itmo.isproject.repository.user

import itmo.isproject.model.user.Role
import itmo.isproject.model.user.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Int> {
    
    fun existsByUsernameInternal(username: String?): Boolean

    fun existsByRole(role: Role?): Boolean

    fun findByUsernameInternal(username: String?): User?

    fun findAllByEnabledFalse(pageable: Pageable): Page<User>

    fun findAllByRole(role: Role?, pageable: Pageable): Page<User>
}
